### PR TITLE
[DCalendar] add unit tests

### PR DESCRIPTION
**Pull request by KayMKM**
### Purpose
*What is the context of this pull request? Why is it being done?*

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

### Tests
*Is this covered by existing tests or new ones? If no, why not?*

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

### CODE REVIEW

File: skills/declarative/extensions/Microsoft.Bot.Solutions.Extensions/Actions/GetContacts.cs [hunk 1]
```diff
@@ -40,7 +42,8 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
             var name = this.NameProperty.GetValue(dcState);
             var token = this.Token.GetValue(dcState);
 
-            var graphClient = GraphClient.GetAuthenticatedClient(token);
+            dc.Context.TurnState.TryGetValue(MockHttpRequestMiddleware.HttpMessageHandlerKey, out var httpHandler);
+            var graphClient = GraphClient.GetAuthenticatedClient(token, (HttpMessageHandler)httpHandler);
```
Comment (xieofxie) [inline skills/declarative/extensions/Microsoft.Bot.Solutions.Extensions/Actions/GetContacts.cs:46]: Will GraphClient.GetAuthenticatedClient create a new one if httpHandler is null?

File: skills/declarative/extensions/Microsoft.Bot.Solutions.Extensions/Actions/GetMeetings.cs [hunk 2]
```diff
@@ -44,7 +46,9 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
             var startTime = StartTime.GetValue(dcState);
             var endTime = EndTime.GetValue(dcState);
 
-            var graphClient = GraphClient.GetAuthenticatedClient(token);
+            dc.Context.TurnState.TryGetValue(MockHttpRequestMiddleware.HttpMessageHandlerKey, out var httpHandler);
+
+            var graphClient = GraphClient.GetAuthenticatedClient(token, (HttpMessageHandler)httpHandler);
```
Comment (xieofxie) [inline skills/declarative/extensions/Microsoft.Bot.Solutions.Extensions/Actions/GetMeetings.cs:51]: Same

File: skills/declarative/extensions/Microsoft.Bot.Solutions.Extensions/Actions/Shared/RetrievePhoto.cs [hunk 3]
```diff
@@ -40,7 +42,8 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
         {
             var dcState = dc.State;
             var token = Token.GetValue(dcState);
-            _graphClient = GraphClient.GetAuthenticatedClient(token);
+            dc.Context.TurnState.TryGetValue(MockHttpRequestMiddleware.HttpMessageHandlerKey, out var httpHandler);
+            _graphClient = GraphClient.GetAuthenticatedClient(token, (HttpMessageHandler)httpHandler);
```
Comment (xieofxie) [inline skills/declarative/extensions/Microsoft.Bot.Solutions.Extensions/Actions/Shared/RetrievePhoto.cs:46]: Same

File: skills/declarative/extensions/Microsoft.Bot.Solutions.Extensions/Actions/GetContacts.cs (hunk 1 above)
Comment (KayMKM) [reply skills/declarative/extensions/Microsoft.Bot.Solutions.Extensions/Actions/GetContacts.cs:46]: yes. the default value of httpHandler property is null

**Pull request merged**